### ISSUE TITLE
Reading .buffer readUntil: 'eof' returns buffer 1 too short

### ISSUE THREAD

**Issue by anajavi** [keichi/binary-parser#32]
Thanks for the library, really helpful!

I've noticed that parsing rest of message into a buffer, it comes back one too short.

```
var Parser = require('binary-parser').Parser;
var bp = new Parser().buffer('test',{ readUntil: 'eof'});
var buf=new Buffer([1,2,3]);

console.log(bp.parse(buf));
```

output:

```
{ test: <Buffer 01 02> }
```

I am not an expert but I think that there should not be - 1 on this line :

```
ctx.pushCode('{0} = buffer.slice(offset, buffer.length - 1);',
```

https://github.com/keichi/binary-parser/blob/d7d2fd15eff3100af317f4264cd51982c8cc0dc3/lib/binary_parser.js#L442

**Comment by jayalfredprufrock** [comment]
This is definitely a bug :( I'm gonna have to use a fork until this is fixed. I'll submit a pull request but it looks like there are some already backlogged.. Either way, thanks for a great library.

**Comment by kkarczmarczyk** [comment]
You can use:

`.array('test', {type: 'uint8', readUntil: 'eof', formatter: (a) => Buffer(a)})`

as temporary workaround.

**Comment by NorthernMan54** [comment]
Tks for the workaround.

**Comment by kertal** [comment]
created a pull request for that 
https://github.com/keichi/binary-parser/pull/47

**Comment by keichi** [comment]
This is now fixed in release 1.1.6 thanks to @kertal.
